syntax = "proto3";

package deduction;

import "google/protobuf/timestamp.proto";

message Deduction {
    uint64 id = 1;                               // 服务 ID
    string name = 2;                             // 服务名称. 如 "滴滴出行微信免密支付"
    string provider = 3;                         // 服务提供商. 如 "滴滴出行"
    string description = 4;                      // 服务详情
    string payment = 5;                          // 扣款方式. 包括 "零钱"
    uint32 status = 6;                           // 服务状态. 包括 0-"未签约", 1-"生效中"
    google.protobuf.Timestamp opened_at = 7;     // 服务开通时间
    google.protobuf.Timestamp closed_at = 8;     // 服务关闭时间
    google.protobuf.Timestamp updated_at = 9;    // 服务更新时间. 如更换扣款方式
    google.protobuf.Timestamp deleted_at  = 10;  // 服务删除时间.
}

message Deductions {
    repeated Deduction deduction = 1;
}

message Request {
}

message Response {
}

service DeductionService {
    rpc GetDeductionServices (Request) returns (stream Deductions) {}
    rpc OpenDeduction (Deduction) returns (Deduction);
    rpc CloseDeduction (Deduction) returns (Deduction);
}